test/backend/lazy_load_test.rb

```diff
@@ -0,0 +1,107 @@
+require 'test_helper'
+
+class I18nBackendLazyLoadTest < I18n::TestCase
+  def setup
+    super
+    @backend  = I18n.backend = I18n::Backend::LazyLoad.new
+    I18n.load_path = [locales_dir + '/en.yml', locales_dir + '/fr.yml']
+  end
+
+  test "only loads translations for current locale" do
+    @backend.reload!
+
+    assert_nil translations
+
+    I18n.with_locale(:en) { @backend.eager_load! }
```

Comment:
I'm kinda torn here.

I think this backend should either raise on `eager_load!` or either load all locales because it's not supposed to be used in eager loading contexts.

That being said our CI is eager loading (and most should be https://github.com/rails/rails/pull/43508).

I'm just very worried people might accidentally use this in production, which would be bad.

Reply:
I think I'd opt to raise on `eager_load!` to make it clear this is not intended to be used in production / other eager loading contexts. If CI is eager loaded the performance benefits would be negligible anyways, no? (And actually probably worse than with a `SimpleBackend`, because of the extra string comparison cost). Just as was the case with us implementing this in Core, the benefits to this backend are reaped mainly for local testing.

Reply:
Another option would be a loud warning, but yeah I also wondered about `eager_load!`. Raising seems reasonable to me.

Reply:
I agree, and thanks all for the feedback—I've altered the behaviour to now raise on `eager_load!`.